File: problems/3349_adjacent_increasing_subarrays_detection_i/3349_adjacent_increasing_subarrays_detection_i.java
class Solution {
    public boolean hasIncreasingSubarrays(List<Integer> nums, int k) {
        int incr = 1;
        int pre = 0;
        int n = nums.size();

        for (int i = 1; i < n; i++) {
            if (nums.get(i) > nums.get(i - 1)) {
                incr++;
            } else {
                pre = incr;
                incr = 1;
            }
            if (incr / 2 >= k || Math.min(pre, incr) >= k) {
                return true;
            }
        }
        return false;
    }
}

File: problems/3349_adjacent_increasing_subarrays_detection_i/3349_adjacent_increasing_subarrays_detection_i.py
class Solution:
    def hasIncreasingSubarrays(self, nums: List[int], k: int) -> bool:
        incr = 1
        pre = 0
        n = len(nums)

        for i in range(1, n):
            if nums[i] > nums[i - 1]:
                incr += 1
            else:
                pre = incr
                incr = 1
            # 檢查條件
            if incr // 2 >= k or min(pre, incr) >= k:
                return True
        return False